media-sound/ezstream/ezstream-1.0.2.ebuild

```diff
@@ -1,31 +1,46 @@
-# Copyright 1999-2024 Gentoo Authors
+# Copyright 1999-2025 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=8
 
+inherit autotools
+
 DESCRIPTION="A command line source client for Icecast media streaming servers"
 HOMEPAGE="https://www.icecast.org/ezstream/"
-SRC_URI="http://downloads.xiph.org/releases/${PN}/${P}.tar.gz"
+SRC_URI="https://downloads.xiph.org/releases/${PN}/${P}.tar.gz"
 
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~amd64 ~ppc ~x86"
-IUSE="taglib"
+IUSE="test"
+RESTRICT="!test? ( test )"
 
 DEPEND="
 	dev-libs/libxml2
 	>=media-libs/libshout-2.2
-	!taglib? ( media-libs/libvorbis )
-	taglib? ( media-libs/taglib:= )"
+	media-libs/taglib:="
 RDEPEND="
 	${DEPEND}
 	net-misc/icecast"
-BDEPEND="virtual/pkgconfig"
+BDEPEND="
+	virtual/pkgconfig
+	test? ( dev-libs/check )"
+
+PATCHES=(
+	"${FILESDIR}/${P}-conditional-check.patch"
+	"${FILESDIR}/${P}-basename-in-libgen.patch"
+)
+
+src_prepare() {
+	default
+	# patching mandatory dependency on libcheck from configure.ac
+	eautoreconf
+}
 
 src_configure() {
 	econf \
 		--enable-examplesdir='$(docdir)/examples' \
-		$(use_with taglib taglib "${ESYSROOT}"/usr)
```

Comment:
Should we pass --with-taglib as well, to be explicit?

Reply:
https://gitlab.xiph.org/xiph/ezstream/-/blob/develop/configure.ac?ref_type=heads#L172-L177
Check is gone, it's unconditional dependency. Passing `--with-taglib` causes build failure, because it expects path to taglib.

Reply:
And I don't want to throw redundant pkg-config invocation into ebuild.